game/settings.py

```diff
@@ -16,78 +16,81 @@ class AutoAtoBehavior(Enum):
 
 @dataclass
 class Settings:
+    version: Optional[str] = None
 
     # Difficulty settings
+    # AI Difficulty
     player_skill: str = "Good"
     enemy_skill: str = "Average"
-    ai_pilot_levelling: bool = True
     enemy_vehicle_skill: str = "Average"
-    map_coalition_visibility: ForcedOptions.Views = ForcedOptions.Views.All
-    labels: str = "Full"
-    only_player_takeoff: bool = True  # Legacy parameter do not use
-    night_disabled: bool = False
-    external_views_allowed: bool = True
-    supercarrier: bool = False
-    generate_marks: bool = True
-    manpads: bool = True
-    version: Optional[str] = None
     player_income_multiplier: float = 1.0
     enemy_income_multiplier: float = 1.0
+    invulnerable_player_pilots: bool = True
+    # Mission Difficulty
+    night_disabled: bool = False
+    manpads: bool = True
+    # Mission Restrictions
+    labels: str = "Full"
+    map_coalition_visibility: ForcedOptions.Views = ForcedOptions.Views.All
+    external_views_allowed: bool = True
+    battle_damage_assessment: bool = True
```

Comment:
Make this a tri-state of enforced on, enforced off, and no enforcement, with no-enforcement as the default? We don't have a type for that, so `Optional[bool]` or make an enum.

(Haven't reviewed fully, but the rest looks fine at a glance, will look more later)

Reply:
How about "Allow battle damage assessment" option like "Allow external view"? Making unchecked enforce battle damage assessment restriction false.

I also made tristate version, too: https://github.com/pbzweihander/dcs_liberation/tree/bda-tristate

Reply:
I like the tri-state version more. It looks good as-is. I went ahead and cherry-picked it, so closing the PR as resolved. I tweaked the names in the options a little bit to improve clarity.

It's a little bit different than "allow external view". If the mission forces "allow external view" players that don't want it can self-enforce by not pushing F2. There's no way for players to avoid BDA messages if the mission forces them on.